app/src/main/java/com/example/dispatcher/MainActivity.kt

```diff
@@ -1,21 +1,68 @@
 package com.example.dispatcher
 
 import android.os.Bundle
-import androidx.activity.ComponentActivity
-import androidx.activity.compose.setContent
-import androidx.compose.material3.MaterialTheme
-import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.ui.tooling.preview.Preview
-
-class MainActivity : ComponentActivity() {
+import android.widget.ImageView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+
+class MainActivity : AppCompatActivity() {
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
         setContentView(R.layout.activity_main)
 
+        val homeTab: ImageView = findViewById(R.id.tab_home)
+        val favoritesTab: ImageView = findViewById(R.id.tab_favorites)
+        val profileTab: ImageView = findViewById(R.id.tab_profile)
+
+        // Set the initial fragment and tab image
+        replaceFragment(HomeFragment())
+        updateTabImages(selectedTab = R.id.tab_home)
+
+        homeTab.setOnClickListener {
+            replaceFragment(HomeFragment())
+            updateTabImages(selectedTab = R.id.tab_home)
+        }
+
+        favoritesTab.setOnClickListener {
+            replaceFragment(FavoritesFragment())
+            updateTabImages(selectedTab = R.id.tab_favorites)
+        }
+
+        profileTab.setOnClickListener {
+            replaceFragment(ProfileFragment())
+            updateTabImages(selectedTab = R.id.tab_profile)
+        }
     }
-}
 
+    private fun replaceFragment(fragment: Fragment) {
+        val fragmentManager = supportFragmentManager
+        val fragmentTransaction = fragmentManager.beginTransaction()
+        fragmentTransaction.replace(R.id.fragmentDisplay, fragment)
+        fragmentTransaction.commit()
+    }
 
+    private fun updateTabImages(selectedTab: Int) {
```

Comment:
there is a feature of Android called 'selector' - which manages the differents states of a component and control the icons.
so you can use it via the XML file instead of doing so programmatically